alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -6,49 +6,89 @@
 
 class kClipManager implements kBatchJobStatusEventConsumer
 {
-
 	const CLIP_NUMBER = 'clipNumber';
 	const LOCK_EXPIRY = 10;
+
 	/**
-	 * @param string $sourceEntryId
-	 * @param entry $clipEntry
+	 * @param kOperationResource $resource
 	 * @param entry $destEntry
-	 * @param $partnerId
-	 * @param array $operationAttributes
-	 * @param int $priority
+	 * @param entry $clipEntry
 	 * @param string $importUrl
+	 * @param int $rootJobId
+	 * @param int $resourceOrder
+	 * @param bool $fillDest
 	 */
-	public function createParentBatchJob($sourceEntryId,$clipEntry, $destEntry, $partnerId, array $operationAttributes, $priority = 0, $importUrl = null)
+	public function createClipConcatJobFromResource($resource, $destEntry, $clipEntry, $importUrl = null, $rootJobId = null, $resourceOrder = null, $fillDest = true)
 	{
+		$partnerId = $clipEntry->getPartnerId();
+		$clipEntryId = $clipEntry->getId();
+		$sourceEntryId = $resource->getResource()->getOriginEntryId();
+		$operationAttributes = $resource->getOperationAttributes();
+
 		$parentJob = new BatchJob();
 		$parentJob->setPartnerId($partnerId);
-		$parentJob->setEntryId($clipEntry->getEntryId());
+		$parentJob->setEntryId($clipEntryId);
+		if($rootJobId)
+		{
+			$parentJob->setRootJobId($rootJobId);
+			$parentJob->setParentJobId($rootJobId);
+		}
 
-		$trackEntry = new TrackEntry();
-		$trackEntry->setEntryId($destEntry->getEntryId());
-		$trackEntry->setTrackEventTypeId(TrackEntry::TRACK_ENTRY_EVENT_TYPE_CLIP);
-		$trackEntry->setDescription("source entry id: [" .$sourceEntryId. "],  template entry id: [" .$clipEntry->getEntryId(). "].");
-		TrackEntry::addTrackEntry($trackEntry);
+		$description = "source entry id: [$sourceEntryId],  template entry id: [$clipEntryId]";
+		$this->addClipTrackEntry($destEntry->getId(), $description);
 
 		$jobData = new kClipConcatJobData($importUrl);
 		if(!$jobData->getImportNeeded())
 		{
-			$this->setDummyOriginalFlavorAssetReady($clipEntry->getId());
+			$this->setDummyOriginalFlavorAssetReady($clipEntryId);
+		}
+		if($resourceOrder !== null)
+		{
+			// clip concat order for multi clip concat
+			$jobData->setResourceOrder($resourceOrder);
 		}
-		$jobData->setDestEntryId($destEntry->getEntryId());
-		$jobData->setTempEntryId($clipEntry->getEntryId());
 
-		//if it is replace(Trim flow) active the copy to destination consumers
-		$this->fillDestEntry($destEntry, $sourceEntryId, $operationAttributes);
+		$jobData->setDestEntryId($destEntry->getId());
+		$jobData->setTempEntryId($clipEntryId);
+
+		if($fillDest)
+		{
+			//if it is replace(Trim flow) active the copy to destination consumers
+			$this->fillDestEntry($destEntry, $sourceEntryId, $operationAttributes);
+		}
 
 		$jobData->setSourceEntryId($sourceEntryId);
 		$jobData->setPartnerId($partnerId);
-		$jobData->setPriority($priority);
-
+		$jobData->setPriority(0);
```

Comment:
why priority is 0?

Reply:
the default priority is 0 and in all usages for this func when passed it is passed as 0